### ISSUE TITLE
"Cannot read file" with lowercase name in mac

### ISSUE THREAD

**Issue by ymoran00** [typescript-eslint/typescript-eslint#2213]
I'm using eslint on mac. I have a large React project in js, and I'm trying to start using ts. I've changed only one file to ts (in my shared/utils/ folder) and tested eslint.
```
eslint --ignore-path .gitignore --ext .js,.ts,.tsx "shared/utils/
```
Expected to pass, but got this error:
```
/Users/yoav/dev/loco/shared/utils/graphQl/makeFlattenGqlResponse.ts
  0:0  error  Parsing error: Cannot read file '/users/yoav/dev/loco'

✖ 1 problem (1 error, 0 warnings)
```

What's weird is that `/Users/*` folder became `/users/*` folder - so I assume this is what caused the file to be not found.

This is what running with --debug showed me:
```
2020-06-11T12:20:01.906Z eslint:file-enumerator Yield: makeFlattenGqlResponse.ts
2020-06-11T12:20:01.906Z eslint:cascading-config-array-factory Load config files for /Users/yoav/dev/loco/shared/utils/graphQl.
2020-06-11T12:20:01.906Z eslint:cascading-config-array-factory Cache hit: /Users/yoav/dev/loco/shared/utils/graphQl.
2020-06-11T12:20:01.906Z eslint:cli-engine Lint /Users/yoav/dev/loco/shared/utils/graphQl/makeFlattenGqlResponse.ts
2020-06-11T12:20:01.906Z eslint:linter Linting code for /Users/yoav/dev/loco/shared/utils/graphQl/makeFlattenGqlResponse.ts (pass 1)
2020-06-11T12:20:01.906Z eslint:linter Verify
2020-06-11T12:20:01.906Z eslint:linter With ConfigArray: /Users/yoav/dev/loco/shared/utils/graphQl/makeFlattenGqlResponse.ts
2020-06-11T12:20:11.518Z eslint:linter Parsing error: Cannot read file '/users/yoav/dev/loco'.
Error: Cannot read file '/users/yoav/dev/loco'.
    at Object.diagnosticReporter [as onUnRecoverableConfigFileDiagnostic] (/Users/yoav/dev/loco/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createWatchProgram.js:95:11)
    at Object.getParsedCommandLineOfConfigFile (/Users/yoav/dev/loco/node_modules/typescript/lib/typescript.js:27944:18)
    at parseConfigFile (/Users/yoav/dev/loco/node_modules/typescript/lib/typescript.js:105948:43)
    at Object.createWatchProgram (/Users/yoav/dev/loco/node_modules/typescript/lib/typescript.js:105618:13)
    at createWatchProgram (/Users/yoav/dev/loco/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createWatchProgram.js:264:22)
    at Object.getProgramsForProjects (/Users/yoav/dev/loco/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createWatchProgram.js:176:30)
    at Object.createProjectProgram (/Users/yoav/dev/loco/node_modules/@typescript-eslint/typescript-estree/dist/create-program/createProjectProgram.js:27:74)
    at getProgramAndAST (/Users/yoav/dev/loco/node_modules/@typescript-eslint/typescript-estree/dist/parser.js:73:32)
    at Object.parseAndGenerateServices (/Users/yoav/dev/loco/node_modules/@typescript-eslint/typescript-estree/dist/parser.js:363:30)
    at Object.parseForESLint (/Users/yoav/dev/loco/node_modules/@typescript-eslint/parser/dist/parser.js:47:51)
2020-06-11T12:20:11.518Z eslint:linter Generating fixed text for /Users/yoav/dev/loco/shared/utils/graphQl/makeFlattenGqlResponse.ts (pass 1)
2020-06-11T12:20:11.518Z eslint:source-code-fixer Applying fixes
2020-06-11T12:20:11.518Z eslint:source-code-fixer shouldFix parameter was false, not attempting fixes
```

Most of the eslint config (`.eslintrc.js`) is this:
```javascript
module.exports = {
  parser: 'babel-eslint',
  extends: ['airbnb', 'prettier', 'prettier/react'],
  plugins: ['prettier', 'redux-saga', 'react', 'react-hooks', 'jsx-a11y', 'local'],
  env: {
    jest: true,
    browser: true,
    node: true,
    es6: true,
  },
  parserOptions: {
    ecmaVersion: 6,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
    },
  },
  rules: {
    // Base rules
    'prettier/prettier': 2,
    'arrow-body-style': [2, 'as-needed'],
    'max-len': 0,
    'newline-per-chained-call': 0,
    'no-console': npmLintMode ? [1, { allow: ['error'] }] : 0,
    'no-unused-vars': 2,
    'require-yield': 2,
    'require-await': 2,
    'prefer-template': 2,
    'no-warning-comments': 1,
    'max-lines-per-function': [1, { max: 120, skipBlankLines: true, skipComments: true }],
    'max-lines': [1, { max: 400, skipBlankLines: true, skipComments: true }],
    camelcase: [
      2,
      {
        allow: ['^UNSAFE_'],
        properties: 'never',
      },
    ],
    // React rules
    'react/destructuring-assignment': 0,
    'react/forbid-prop-types': [2, { forbid: ['any'] }],
    'react/jsx-filename-extension': 0,
    'react/jsx-first-prop-new-line': [2, 'multiline'],
    'react/jsx-no-target-blank': 0,
    'react/jsx-uses-vars': 2,
    'react/no-array-index-key': 0,
    'react/no-typos': 2,
    'react/prefer-stateless-function': 0,
    'react/require-default-props': 0,
    'react/require-extension': 0,
    'react/self-closing-comp': 2,
    'react/sort-comp': 0,
    'react/jsx-props-no-spreading': 0,
    'react/static-property-placement': 0,
    'react/state-in-constructor': 0,
    'react-hooks/rules-of-hooks': 2,
    'react-hooks/exhaustive-deps': 0,


    // Redux rules
    'redux-saga/no-yield-in-race': 2,
    'redux-saga/yield-effects': 2,

    // Homegrown rules
    'local/progress-reporter': [1, { printSummaryOnExit: npmLintMode }],
    'local/prefer-nullish-coalescing-operator': 1,

    // Plugin rules
    /**
     * eslint-plugin-import
     *
     * @see https://github.com/benmosher/eslint-plugin-import
     */
    'import/no-named-as-default': 0,
    'import/no-anonymous-default-export': 2,
    'import/no-unresolved': 2,
    'import/no-webpack-loader-syntax': 2,
    'import/prefer-default-export': 0,
    'import/named': 1,
    'import/default': 1,
    'import/no-absolute-path': 1,
    'import/no-dynamic-require': 0,
    'import/no-self-import': 1,
    'import/no-useless-path-segments': 1,
    // Helpful warnings
    'import/no-mutable-exports': 1,
    // Style guide
    'import/first': 1,
    'import/no-duplicates': 1,
    'import/extensions': [
      'error',
      'ignorePackages',
      {
        js: 'never',
        jsx: 'never',
        ts: 'never',
        tsx: 'never',
      },
    ],
    'import/order': [
      2,
      {
        groups: [
          'builtin',
          'external',
          'internal',
          'parent',
          'sibling',
          'index',
          'unknown',
        ],
        
        'newlines-between': 'always',
      },
    ],
    'import/newline-after-import': 1,
    'import/no-default-export': 0,
    'import/group-exports': 0,
  },
  overrides: [
    {
      files: '**/*.+(ts|tsx)',
      parser: '@typescript-eslint/parser',
      parserOptions: {
        project: [
          './tsconfig.json',
          ...appsMapping.map((app) => `./apps/${app.name}/tsconfig.json`).join(', '),
        ],
        debugLevel: 'typescript-eslint',
      },
      plugins: ['@typescript-eslint/eslint-plugin'],
      extends: [
        // Disable some of the rules typescript renders unnecessary (duplication between eslint)
        'plugin:@typescript-eslint/eslint-recommended',
        // Some best practices for handling ts interfaces and such
        // See: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.json
        'plugin:@typescript-eslint/recommended',
        // Make typescript respect prettier styling defs
        'eslint-config-prettier/@typescript-eslint',
      ],
    },    
  ],
  settings: {
    // https://github.com/benmosher/eslint-plugin-import#resolvers
    'import/resolver': {
      alias: {
        map: [['@shared', './shared']],
        extensions: ['.js', '.jsx', '.ts', '.tsx', '.json'],
      },
    },
  },
};

```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `3.2.0` |
| `TypeScript`                | `3.9.5` |
| `ESLint`                    | `7.2.0` |
| `node`                      | `12.6.2` |
| `npm`                       | `6.14.4` |

**Comment by bradzacher** [comment]
macos is case **IN**sensitive for filenames, which is why typescritpt lowercases the paths.

I'm not able to repro this - could you please provide a repro repo to help investigate?

**Comment by ymoran00** [comment]
Solved it thanks to the idea to create repro repo. When I succeeded recreating it I discovered the problem...
It was in the eslintrc portion that I omitted because it was related to internal code....
I had this code, using an object called appsMapping that has the names of all apps in our repo:

```JSON
...
overrides: [
    {
      files: '**/*.+(ts|tsx)',
      parser: '@typescript-eslint/parser',
      parserOptions: {
        project: [
          './tsconfig.json',
          ...appsMapping.map((app) => `./apps/${app.name}/tsconfig.json`).join(', '),
        ],
      },
      plugins: ['@typescript-eslint/eslint-plugin'],
}
...
```
What I haven't noticed is that we did there join to a string, and then spread operator on the string... The path was invalid.
